js/main.js

```diff
@@ -14,11 +14,135 @@
+    var pinCoordinates = mainPin.getBoundingClientRect();
+    var pinEdge = window.getComputedStyle(mainPin, ':after');
+    var pinEdgeHeight = parseInt(pinEdge.height, 10);
+    return 'x: ' + Math.floor(pinCoordinates.x + pinCoordinates.width / 2) + '; y: ' + Math.floor(pinCoordinates.y + pinCoordinates.height + pinEdgeHeight);
+  }
+
+  // Коллбэк, убирает класс с карты, рендерим пины, убираем атрибуты disabled, снимаем слушателя с mainPin
+  function initMapActiveState() {
+    var addressInput = document.querySelector('#address');
+    map.classList.remove('map--faded');
+    form.classList.remove('ad-form--disabled');
+    controlsRemoveAttribute(formInputElements);
+    controlsRemoveAttribute(formSelectElements);
+    mapFilters.removeAttribute('disabled');
+    formTextarea.removeAttribute('disabled');
+    formSubmit.removeAttribute('disabled');
+    generatePins(createMocksForData(OFFERS_NUMBER));
+    mainPin.removeEventListener('mousedown', mainPinMousedownHandler);
+    mainPin.removeEventListener('keydown', mainPinKeydownHandler);
+    addressInput.value = setPinCoordinates();
+  }
+
+  // Создаем функцию валидации соответствия количеству комнат количества гостей
+  function compareNumberOfRoomsWithNumberOfGuests() {
+    if (roomNumberValue.value === '1' && guestsNumber.value === '1') {
+      validationMark = true;
+      roomNumberValue.setCustomValidity('');
+    } else {
+      roomNumberValue.setCustomValidity('Извините, данные апартаменты предназначены для одного гостя, выберите другой вариант');
+    }
+    if (roomNumberValue.value === '2') {
+      if (guestsNumber.value === '1' || guestsNumber.value === '2') {
+        validationMark = true;
+        roomNumberValue.setCustomValidity('');
+      } else {
+        roomNumberValue.setCustomValidity('Извините, данные апартаменты предназначены для одного или двух гостей, выберите другой вариант');
+      }
+    }
+    if (roomNumberValue.value === '3') {
+      if (guestsNumber.value === '1' || guestsNumber.value === '2' || guestsNumber.value === '3') {
+        validationMark = true;
+        roomNumberValue.setCustomValidity('');
+      } else {
+        validationMark = false;
+        roomNumberValue.setCustomValidity('Извините, данные апартаменты предназначены только для гостей, выберите другой вариант');
+      }
+    }
+    if (roomNumberValue.value === '100') {
+      if (guestsNumber.value === '0') {
+        validationMark = true;
+        roomNumberValue.setCustomValidity('');
+      } else {
+        roomNumberValue.setCustomValidity('Извините, данные апартаменты не предназначены для гостей, выберите другой вариант');
+      }
+    }
+  }
+
+  // Добавляем слушателя на сабмит, если валидация успешна.
+  form.addEventListener('submit', function (evt) {
+    evt.preventDefault();
+    if (validationMark === true) {
+      form.submit();
+    }
+  });
+
+  submitButton.addEventListener('click', compareNumberOfRoomsWithNumberOfGuests);
```

Comment:
```suggestion
  submitButton.addEventListener('click', submitButtonСlickHandler);
```

может тебе сделать через анонимную функцию?